### PR TITLE
Terraform code for Self-Hosted Runners

### PR DESCRIPTION
**Pull request by elink21**
- Adding Terraform CodeBase for Self-Hosted Runners ( Linux and Windows )
- Updated Readme for Self-Hosted Runners

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/get-started-contributing/#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI.

### CODE REVIEW

**Comment by elink21** [comment]
Hi @damccorm , to simplify the management process we've created an IaC base for linux and windows self-hosted runners, also I've updated the documentation according to the new values.

File: .github/gh-actions-self-hosted-runners/README.md [hunk 1]
```diff
@@ -21,29 +21,29 @@ The current GitHub Actions workflows are being tested on multiple operating syst
 On the other hand, we will rely on GitHub-hosted runners for MacOS builds until a straightforward implementation approach comes out.
 
 ## Ubuntu
-Ubuntu Self-hosted runners are stored in Artifact Registry and implemented using Google Kubernetes Engine with the following specifications:
+Ubuntu Self-hosted runners are stored in Container Registry and implemented using Google Kubernetes Engine with the following specifications:
 
 #### Cluster
 * Cluster: [gh-actions-linux-runners](https://console.cloud.google.com/kubernetes/clusters/details/us-central1-a/gh-actions-linux-runners/details?project=apache-beam-testing)
-* Image: [linux-github-actions-runner](https://console.cloud.google.com/artifacts/docker/apache-beam-testing/us-central1/beam-github-actions/linux-github-actions-runner?project=apache-beam-testing)
+* Image: [linux-github-actions-runner](https://console.cloud.google.com/gcr/images/apache-beam-testing/global/ubuntu-runners-self-hosted?project=apache-beam-testing)
```
Comment (Abacn) [inline .github/gh-actions-self-hosted-runners/README.md:28]: just wondering is this change reflecting these containers are moving from artifact registry to container registry? per gcp documentation container registry is in maintenance mode and artifact registry supersedes it. There is some voice to migrate existing container workflows to artifact registry though not yet happened.

**Comment by damccorm** [comment]
stop reviewer notifications

**Comment by damccorm** [comment]
stop reviewer notifications

**Pull request closed without merging**